### PR TITLE
Feature/move header and footer to file

### PR DESCRIPTION
**Pull request by tobi812**

### CODE REVIEW

File: src/PdfGenerator.php [hunk 1]
```diff
@@ -51,40 +52,40 @@ public function __construct($binaryPath)
...
+            $preparedOptions = $this->convertImagesToBase64($preparedOptions, 'footerContent');
+        }
 
         $this->convertToPdf($tmpFilePath, $targetPath, $preparedOptions);
     }
 
     /**
      * @param string $html
-     * @param array|Options|null options
+     * @param array|null $options
```
Comment (florianpreusner) [inline src/PdfGenerator.php]: what about just allowing arrays? cause they also can be empty?

Comment (tobi812) [inline src/PdfGenerator.php]: Well the idea is, that the user can decide, if he wants to use an array or the Options-Object. Anyway the Object is missing in the DocBlock.

Comment (nahpeps) [inline src/PdfGenerator.php]: @tobi812 : why would you like to allow the user to use an array if there is the possibility to have a clear structured  Options Object?

Comment (tobi812) [inline src/PdfGenerator.php]: Well that is a good point actually :) I will change that!

**Comment by florianpreusner** [comment]
One general question: Why is there a Base64ConverterInterface? Is it really possible to have a different implementation? In your DocBlocks you are using the converter (Base64Converter) directly.

**Comment by tobi812** [comment]
@florianpreusner That is true, i have to update those docblocks, if the user prefers another implementation of the Converter he can inject it via setter method.

**Comment by florianpreusner** [comment]
lgtm 👍

**Pull request merged**